zcash_client_backend/src/data_api.rs

```diff
@@ -1542,6 +1621,28 @@ pub trait WalletWrite: WalletRead {
     ///
     /// There may be restrictions on heights to which it is possible to truncate.
     fn truncate_to_height(&mut self, block_height: BlockHeight) -> Result<(), Self::Error>;
+
+    /// Reserves the next `n` available ephemeral addresses for the given account.
+    /// This cannot be undone, so as far as possible, errors associated with transaction
+    /// construction should have been reported before calling this method.
+    ///
+    /// To ensure that sufficient information is stored on-chain to allow recovering
+    /// funds sent back to any of the used addresses, a "gap limit" of 20 addresses
+    /// should be observed as described in [BIP 44].
+    ///
+    /// Returns an error if there is insufficient space within the gap limit to allocate
+    /// the given number of addresses, or if the account identifier does not correspond
+    /// to a known account.
+    ///
+    /// Precondition: `n >= 0`
+    ///
+    /// [BIP 44]: https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#user-content-Address_gap_limit
+    #[cfg(feature = "transparent-inputs")]
+    fn reserve_next_n_ephemeral_addresses(
+        &mut self,
+        account_id: Self::AccountId,
+        n: i32,
```

Comment:
```suggestion
        n: NonZeroU32,
```
Why would this be a signed integer, or zero?

Reply:
It's `i32` because it was convenient to constrain it to be $< 2^{31}$, to avoid a potential overflow, and there is no `u31` or `NonZeroU31` type. But yes it does look a bit odd. Perhaps making the precondition `n < 0x80000000` would be better.

Reply:
I changed it to `u32` with precondition `n < 0x80000000`.